cmd/kubectl-gadget/collector.go

```diff
@@ -141,7 +153,66 @@ func collectorCmdRun(subCommand string) func(*cobra.Command, []string) {
 			w.Flush()
 		}
 	}
-	return func(cmd *cobra.Command, args []string) {
-		utils.GenericTraceCommand(subCommand, &collectorParams, args, "Status", callback, nil)
+
+	utils.GenericTraceCommand("process-collector", &commonParams, args, "Status", callback, nil)
+}
+
+func socketCollectorCmdRun(cmd *cobra.Command, args []string) {
+	callback := func(contextLogger *log.Entry, nodes *corev1.NodeList, results *gadgetv1alpha1.TraceList) {
+		allSockets := []socketcollectortypes.Event{}
+
+		for _, i := range results.Items {
+			var sockets []socketcollectortypes.Event
+			json.Unmarshal([]byte(i.Status.Output), &sockets)
+			allSockets = append(allSockets, sockets...)
+		}
+
+		sort.Slice(allSockets, func(i, j int) bool {
+			si, sj := allSockets[i], allSockets[j]
+			switch {
+			case si.Event.Node != sj.Event.Node:
+				return si.Event.Node < sj.Event.Node
+			case si.Event.Namespace != sj.Event.Namespace:
+				return si.Event.Namespace < sj.Event.Namespace
+			case si.Event.Pod != sj.Event.Pod:
+				return si.Event.Pod < sj.Event.Pod
+			case si.Protocol != sj.Protocol:
+				return si.Protocol < sj.Protocol
+			case si.Status != sj.Status:
+				return si.Status < sj.Status
+			default:
+				return si.LocalAddress < sj.LocalAddress
```

Comment:
Shouldn't you add LocalPort, RemoteAddress and RemotePort too?

Reply:
I just thought that after having them sorted by `LocalAddress` the rest was not that relevant but you are right, if we have the possibility of sorting everything, let's do it. I will add them.